mirascope/core/groq/stream.py

```diff
@@ -28,11 +30,20 @@ class GroqStream(
 ):
     _provider = "groq"
 
+    @property
+    def cost(self) -> float | None:
+        """Returns the cost of the call."""
+        return calculate_cost(self.input_tokens, self.output_tokens, self.model)
+
     def _construct_message_param(
-        self, tool_calls: list | None = None, content: str | None = None
+        self,
+        tool_calls: list[ChatCompletionMessageToolCallParam] | None = None,
+        content: str | None = None,
     ) -> ChatCompletionAssistantMessageParam:
-        return ChatCompletionAssistantMessageParam(
+        message_param = ChatCompletionAssistantMessageParam(
             role="assistant",
             content=content,
-            tool_calls=tool_calls,
         )
+        if tool_calls:
```

Comment:
we pass in None already if tool_calls is falsey. if you want to make this change, i would imagine `_stream.py` in our openai module also needs to change

Reply:
No, because there are some providers that accept `None` and I don't check there, but if we pass in `list[Unknown]` that could cause issues.